src/L2ArbitrumGovernorV2.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import {
+    L2ArbitrumGovernor,
+    GovernorUpgradeable,
+    IGovernorUpgradeable
+} from "src/L2ArbitrumGovernor.sol";
+
+contract L2ArbitrumGovernorV2 is L2ArbitrumGovernor {
+    /// @notice Error thrown when attempting to cancel a proposal that is not in Pending state.
+    /// @param state The current state of the proposal.
+    error ProposalNotPending(GovernorUpgradeable.ProposalState state);
+
+    /// @notice Error thrown when a non-proposer attempts to cancel a proposal.
+    /// @param sender The address attempting to cancel the proposal.
+    /// @param proposer The address of the actual proposer.
+    error NotProposer(address sender, address proposer);
+
+    /// @notice Mapping from proposal ID to the address of the proposer.
+    /// @dev Used in cancel() to ensure only the proposer can cancel the proposal.
+    mapping(uint256 => address) internal proposers;
+
+    /// @notice Creates a new proposal and records the proposer.
```

Comment:
nit: "records the proposer" is not a high level concern of the caller, it's more of an implementation concern. i think here we'd be better off inheriting the IGovernorUpgradeable natspec, if possible